gax/src/main/java/com/google/api/gax/rpc/Watchdog.java

```diff
@@ -56,70 +54,45 @@
  *       stream and forcefully closing the stream. This is measured from the last time the caller
  *       had no outstanding demand. Duration.ZERO disables the timeout.
  * </ul>
- *
- * @param <ResponseT> The type of the response.
  */
 @InternalApi
-public class Watchdog<ResponseT> {
+public class Watchdog implements Runnable {
   // Dummy value to convert the ConcurrentHashMap into a Set
   private static Object VALUE_MARKER = new Object();
```

Comment:
Super-minor: in HashSet implementation this is called `PRESENT` maybe it will be nicer to have it named same way here.